src/api/app/models/package.rb

```diff
@@ -1194,6 +1194,12 @@ def commit( rev = nil )
   end
 
   def self.verify_file!(pkg, name, content)
+    raise IllegalFileName, "'#{name}' is not a valid filename" if (!name.present? || !(name =~ %r{^[^\/]+$}))
+
+    # file is an ActionDispatch::Http::UploadedFile and Suse::Validator.validate
+    # will call to_s therefore we have to read the content first
+    content = File.open(content.path).read if content.kind_of?(ActionDispatch::Http::UploadedFile)
```

Comment:
Suse::Validator.validate calls to_s which fails is for a new file because it's an ActionDispatch::Http::UploadedFile object!